Source/WebKit/GPUProcess/GPUProcessSessionParameters.h

```diff
@@ -41,48 +41,28 @@ struct GPUProcessSessionParameters {
     SandboxExtension::Handle mediaKeysStorageDirectorySandboxExtensionHandle;
 #endif
 
-    template<class Encoder>
-    void encode(Encoder& encoder) const
+    static constexpr auto ipcData()
     {
-        encoder << mediaCacheDirectory << mediaCacheDirectorySandboxExtensionHandle;
+        return std::make_tuple(
+            &GPUProcessSessionParameters::mediaCacheDirectory,
+            &GPUProcessSessionParameters::mediaCacheDirectorySandboxExtensionHandle
 #if ENABLE(LEGACY_ENCRYPTED_MEDIA)
-        encoder << mediaKeysStorageDirectory << mediaKeysStorageDirectorySandboxExtensionHandle;
+            , &GPUProcessSessionParameters::mediaKeysStorageDirectory,
+            &GPUProcessSessionParameters::mediaKeysStorageDirectorySandboxExtensionHandle
 #endif
+            );
+    }
+
+    template<class Encoder>
+    void encode(Encoder& encoder) const
```

Comment:
I'm not entirely fond of adding yet another method for encoding and decoding.
I'd be interested more in approach of 
```
template<class Endecoder>
void endecode(Endecoder& endecoder) DOSOMETHINGWITHCONST {
   endecoder(mediaCacheDirectory)(mediaCacheDirectorySandboxExtensionHandle)(mediaKeysStorageDirectory)(and-so-on);
}
```

Alternatively call `ipcData()` as `codedFields()` or something, and have the Encoder/Decoder templates detect that method only, so that the extra encode/decode boilerplate is not needed.

Reply:
There is also some work being done about generating the encoders/decoders, but unfortunately I haven't followed that one at all..